kanidmd/src/lib/be/mod.rs

```diff
@@ -1227,19 +1199,19 @@ impl<'a> BackendWriteTransaction<'a> {
 
             // Write the changes out to the backend
             if let Some(rem) = n2u_rem {
-                idlayer.write_name2uuid_rem(audit, rem)?
+                idlayer.write_name2uuid_rem(rem)?
             }
 
             match u2s_act {
                 None => {}
-                Some(Ok(k)) => idlayer.write_uuid2spn(audit, uuid, Some(k))?,
-                Some(Err(_)) => idlayer.write_uuid2spn(audit, uuid, None)?,
+                Some(Ok(k)) => idlayer.write_uuid2spn(uuid, Some(k))?,
+                Some(Err(_)) => idlayer.write_uuid2spn(uuid, None)?,
             }
```

Comment:
Not sure why this is rendering with lines from the old + new changes, but you just simplify this with if-let and `Result::ok`.

```suggestion
            if let Some(res) = u2s_act {
                idlayer.write_uuid2spn(uuid, res.ok())?;
            }
```